takari-lifecycle-plugin/src/main/java/io/takari/maven/plugins/compile/AbstractCompileMojo.java

```diff
@@ -433,18 +448,70 @@ public void execute() throws MojoExecutionException, MojoFailureException {
     }
   }
 
+  private void checkDependencyReferences(Set<File> referencedEntries) throws MojoExecutionException, IOException {
+    Set<Artifact> referencedArtifacts = new LinkedHashSet<>();
+    Set<Artifact> undeclaredArtifacts = new LinkedHashSet<>();
+    Set<Artifact> unusedArtifacts = new LinkedHashSet<>();
+    List<File> processorPath = getProcessorpath();
+
+    // Include processor path in referenced entries
+    Set<File> allReferencedEntries = new LinkedHashSet<>();
+    allReferencedEntries.addAll(referencedEntries);
+    if (processorPath != null) {
+      allReferencedEntries.addAll(processorPath);
+    }
+
+    // Find the equivalent artifact equivalent for each referenced entry
+    for (File entry : allReferencedEntries) {
+      Artifact artifact = getClasspath().get(entry);
```

Comment:
This unnecessary calculates new classpath map for each referenced entry. At least pull `#getClasspath()` out of the loop. Better yet, pass classpath map as a parameter, so the same map is used during compilation and when checking for unused dependencies. (which will also ensure the two classpaths are the same)